src/dialectic/core.py

```
@@ -371,14 +373,311 @@ def _log_response_metrics(
+                    rendered_parts.append(
+                        json.dumps(item, ensure_ascii=False, default=str)
+                    )
+                return "\n".join(part for part in rendered_parts if part)
+
+            if isinstance(content, dict):
+                return json.dumps(
+                    cast(dict[str, Any], content), ensure_ascii=False, default=str
+                )
+
+            return str(cast(object, content))
+        except Exception:
```

Ruff (0.14.14)

Do not catch blind exception: `Exception`

(BLE001)